test/cases/generate-critical-css/generate-critical-css.spec.js

```diff
@@ -5,54 +5,71 @@ const { JSDOM } = jsdom;
-        
-        assert.equal(inlineStyleTags.length, 1);
-        assert.equal(inlineStyleTags[0].getAttribute('type'), 'text/css');
+        indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
+        indexHtmlDom = new JSDOM(indexHtmlString);
 
         done();
       });
     });
 
-    it('should generate the expected critical <link> tag', (done) => {
-      webpack(webpackConfig, () => {
-        const indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
-        const indexHtmlDom = new JSDOM(indexHtmlString);
-        const linkTags = indexHtmlDom.window.document.querySelectorAll('link');
-        
-        assert.equal(linkTags.length, 2);
-        assert.equal(linkTags[0].getAttribute('rel'), 'preload');
-        assert.equal(linkTags[0].getAttribute('as'), 'style');
+    it('should generate the expected critical inline <style> tag', () => {
+      const inlineStyleTags = indexHtmlDom.window.document.querySelectorAll('style');
 
-        assert.equal(linkTags[1].getAttribute('rel'), 'preload');
-        assert.equal(linkTags[1].getAttribute('as'), 'style');
+      assert.equal(inlineStyleTags.length, 1);
+      assert.equal(inlineStyleTags[0].getAttribute('type'), 'text/css');
+    });
 
-        done();
-      });
+    it('should generate the expected critical <link> tag', () => {
+      const linkTags = indexHtmlDom.window.document.querySelectorAll('link');
+      
+      assert.equal(linkTags.length, 2);
+      assert.equal(linkTags[0].getAttribute('rel'), 'preload');
+      assert.equal(linkTags[0].getAttribute('as'), 'style');
+
+      assert.equal(linkTags[1].getAttribute('rel'), 'preload');
+      assert.equal(linkTags[1].getAttribute('as'), 'style');
     });
 
-    it('should generate the expected critical <noscript> tag', (done) => {
-      webpack(webpackConfig, () => {
-        const indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
-        const indexHtmlDom = new JSDOM(indexHtmlString);
-        const noscriptTags = indexHtmlDom.window.document.querySelectorAll('noscript');
-        
-        assert.equal(noscriptTags.length, 2);
+    it('should generate the expected critical <noscript> tag', () => {
+      const noscriptTags = indexHtmlDom.window.document.querySelectorAll('noscript');
+      
+      assert.equal(noscriptTags.length, 2);
+    });
+
+  });
+
+  describe('public path', () => {
+    let indexHtmlString;
+    let indexHtmlDom;
 
+    const config = merge({ output: { publicPath: '//cdn.example.com/' } }, webpackConfig);
+
+    before((done) => {
+      webpack(config, () => {
+        indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
+        indexHtmlDom = new JSDOM(indexHtmlString);
+        
         done();
       });
     });
 
+    it('should generate the expected critical inline <style> tag', () => {
```

Comment:
without getting too into the weeds on this, do you think there is something unique to this feature being added we can test for?  Maybe just a string search that the `publicPath` is present in the built _index.html?  This would at least give confidence that the logic added to support this "works" and that this test doesn't just give us a false positive.  (.e.g the `filter` / `map` / `join` logic is doing what it should be doing)

open to thoughts / feedback.

cc: @anthonygore